src/System.Private.CoreLib/src/System/Environment.cs

```diff
@@ -76,7 +76,11 @@ public static int CurrentManagedThreadId
         {
             get
             {
+#if PLATFORM_UNIX
```

Comment:
We prefer to use C# partial classes in separate files for platform specific code. The whole implementation of CurrentManagedThreadId should be moved to Environment.Windows.cs and Environment.Unix.cs, with no #ifdefs.